utility/common.go

```diff
@@ -13,3 +13,27 @@ func CleanUrlPath(p string) string {
 	url.Path = path.Clean(url.Path)
 	return url.String()
 }
+
+// InsertInMap modifies `data` *in place*, inserting `values` at the given `key` or at the top level if the `key`is "".
+// If `key` doesn't exist, it gets created.
+// If `key` exists and its value is not a map, InsertInMap does nothing.
+func InsertInMap(data map[string]interface{}, key string, values map[string]interface{}) {
```

Comment:
I think this should be possible already when using `common.MapStr`'s `Put` method.

Reply:
I tried to replicate the behaviour copying the test for this with common.MapStr and the test fails pretty much every case as it doesn't expand the map in the way I need to.

This implementation is a mental leftover from a previous attempt, so Im changing this to be much simpler.